src/librustdoc/config.rs

```diff
@@ -448,6 +454,9 @@ impl Options {
         let generate_search_filter = !matches.opt_present("disable-per-crate-search");
         let persist_doctests = matches.opt_str("persist-doctests").map(PathBuf::from);
         let generate_redirect_pages = matches.opt_present("generate-redirect-pages");
+        let codegen_options_strs = matches.opt_strs("C");
+        let lib_strs = matches.opt_strs("L");
+        let extern_strs = matches.opt_strs("extern");
```

Comment:
Hm, I wouldn't expect us to need to add more options to the rustdoc CLI; if these used to never be set we should continue to not read them (and just rely on default values and such).

Reply:
this do not add new parameters I only fetch them again, the problem was to get back to the string representation of the parameters to send in to rustc.

as the rustdoc lib.rs have this
`stable("C", |o| { o.optmulti("C", "codegen", "pass a codegen option to rustc", "OPT[=VALUE]")`
I feel that I need to get the strings that was sent in to send to rustc

Reply:
So long as the surface area of rustdoc's accepted options does not increase, then this is fine, otherwise we'll want to re-stringify the options as needed from what is currently accepted. For example, we want to be sure that `-C foo` isn't accepted after this PR.

Reply:
`-C foo` will not be accepted by rustdoc as the `let codegen_options = build_codegen_options(matches, error_format);` still is done for the in process rustc call that finds all the tests. that call is in to rustc option parser so as it is all codegen options was and accepted